log/context.go

```diff
@@ -15,35 +17,49 @@ const (
 	traceIDKey = "traceID"
 )
 
-// AttachArgs are used to create loggers to be attached to context object with
-// pre-filled key-value pairs.
+// AttachArgs are used to create loggers and sentry hubs to be attached to
+// context object with pre-filled key-value pairs.
 //
 // All zero value fields will be ignored and only non-zero values will be
 // attached.
 //
 // AdditionalPairs are provided to add any free form, additional key-value pairs
-// you want to attach to all logs from the same context object.
+// you want to attach to all logs and sentry reports from the same context
+// object.
 type AttachArgs struct {
 	TraceID string
 
 	AdditionalPairs map[string]interface{}
 }
 
-// Attach attaches a logger with data extracted from args into the context
-// object.
+// Attach attaches a logger and sentry hub with data extracted from args into
+// the context object.
 func Attach(ctx context.Context, args AttachArgs) context.Context {
-	// Number of non-AdditionalPairs fields in AttachArgs struct.
-	const additional = 1
-	kv := make([]interface{}, 0, len(args.AdditionalPairs)*2+additional)
+	// create and attach the sentry hub
+	hub := sentry.GetHubFromContext(ctx)
+	if hub == nil {
+		hub = sentry.CurrentHub()
+	}
+	hub = hub.Clone()
+	hub.ConfigureScope(func(scope *sentry.Scope) {
+		if args.TraceID != "" {
+			scope.SetTag("trace_id", args.TraceID)
+		}
+		for k, v := range args.AdditionalPairs {
+			scope.SetTag(k, fmt.Sprintf("%v", v))
+		}
+	})
+	ctx = context.WithValue(ctx, sentry.HubContextKey, hub)
 
+	// create and attach the logger
+	const additional = 1 // Number of non-AdditionalPairs fields in AttachArgs struct.
+	kv := make([]interface{}, 0, len(args.AdditionalPairs)*2+additional)
```

Comment:
shouldn't this be 2*(len(AdditionalPairs)+1) since it needs to add on a {key, value} pair, which is two "values"?